doc/how-to-analyze-haskell-code-performance.rst

```diff
@@ -0,0 +1,126 @@
+How to analyze Haskell performance
+==================================
+
+When a Haskell application is slow or uses too much memory,
+Cabal and `GHC <https://downloads.haskell.org/ghc/latest/docs/users_guide/>`__
+can help you understand why.
+
+The main steps are to
+
+1. let GHC insert performance measuring code into your application,
+2. run the application to collect a performance report and
+3. visualize that report.
+
+The process of inserting performance measuring code and collecting performance information
+is called "profiling". The main work is done by GHC while Cabal only acts
+as a build configuration interface and passes corresponding compiler flags to GHC.
+
+Profiling CPU performance
+-------------------------
+
+The first step to build your application, e.g. ``my-app``, with profiling enabled, and
+the second step to run it to collect a report, can be done with a single ``cabal run`` command:
+
+.. code-block:: console
+
+      $ cabal run --enable-profiling --profiling-detail=late my-app -- +RTS -pj -RTS
+      <program runs and finishes>
```

Comment:
I would mention that there are two outputs, in particular `-p` and `-pj` and that you will, just for educational purposes, show the `-pj` in particular.

Reply:
From my comment below, I would actually mention the three if we are going to mention any at all.

Reply:
Yes, please! The eventlog property that Javier is getting at is called time series, I believe? https://en.wikipedia.org/wiki/Time_series

Reply:
All three mentioned.